integration_tests/src/main/python/data_gen.py

```diff
@@ -651,6 +651,19 @@ def _gen_random(self, rand):
     def start(self, rand):
         self._start(rand, lambda: self._gen_random(rand))
 
+class BinaryGen(DataGen):
+    """Generate BinaryType values"""
+    def __init__(self, min_length=0, max_length=20, nullable=True):
+        super().__init__(BinaryType(), nullable=nullable)
+        self._min_length = min_length
+        self._max_length = max_length
+
+    def start(self, rand):
+        def gen_bytes():
+            length = rand.randint(self._min_length, self._max_length)
+            return bytearray([ rand.randint(0, 255) for _ in range(length) ])
```

Comment:
This should use `bytes` instead of `bytearray` since the former is the immutable version.
```suggestion
            return bytes([ rand.randint(0, 255) for _ in range(length) ])
```